nautobot/utilities/testing/__init__.py

```diff
@@ -32,15 +42,55 @@
     "ModelTestCase",
     "ModelViewTestCase",
     "ViewTestCases",
+    "run_job_for_testing",
 )
 
 
+def run_job_for_testing(job, data=None, commit=True, username="test-user", request=None):
+    """Provide a common interface to run Nautobot jobs as part of unit tests."""
+    if data is None:
+        data = {}
+    user_model = get_user_model()
+    user_instance, _ = user_model.objects.get_or_create(username=username, is_superuser=True, password="password")
```

Comment:
Ideally, if there's a provided `request`, we should default to using `request.user` (if any) rather than looking up/creating a distinct User object.